### DISCUSSION TITLE
Video stream crashes eventually on Wyze V2

### DISCUSSION THREAD

**Discussion by clee604** [gtxaspec/wz_mini_hacks#164] [Development]
My problematic V2 continues to give me grief (first with ethernet not working in new versions) and now it seems the video stream crashes after some time. This occurs even with older versions (I tested up until the last version ethernet worked for me, https://github.com/gtxaspec/wz_mini_hacks/tree/66031b41f53ad5578c31c2b9a56e52211fdac85d).

I suspect the RTSP stream isn't being closed when a client disconnects and eventually it gets overloaded and crashes. I can still access the camera via SSH, and the Wyze app can still access the settings (ie. can restart camera) but no video feed (it stalls at getting video stream).

There anything I can do to further debug this since I can still access SSH when the video feed hangs? I noticed on the sd card, while the video feed stopped working, it is still recording these files of consistent file sizes.
![image](https://user-images.githubusercontent.com/93049965/177618253-dd080736-077c-4e98-a74f-b60f0497739f.png)

I thought maybe it was the fat32 file system issue, but I ran the sd card through disk scan and no errors were detected on this sd card. I've tried different sd cards as well to see if that was the issue but no dice...

The wyze RTSP firmware was working 24/7 so I don't think the camera is defective, but I rather not run an old firmware unless I have to.

**Comment by gtxaspec** [comment]
can you provide more information about your setup?  the V2's that I have, run really well, and I even have a few broadcasting to twitch and they experience no issues.

Firmware Version
wz_mini release date
sd card size/manufacturer
continuous recording to sd card?
program you are recording with via rtsp

**Comment by clee604** [comment]
Firmware: demo_4.9.8.1002
wz_mini release date - currently running newest release (July 5)
SD Card - i've been testing with three cards, 2x a32gb kingston canvas select, 64gb WD purple

SD card is set to events only,
all notifications disabled, 
detection zone off, 
all event recording is off (it still records events to SD, not sure if this is intended or setting is for cloud only), 
smart detection off, 
alarm settings all off, 
night vision off. 
camera status light off,
rotate 180 off,
show timestamp on,
show wyze logo off,
record sound on

I'm not doing 24/7 recording but rather using simple-rtsp-server (set to stream only on demand) to proxy the stream to android tinycam clients. This is done so if there are multiple clients, technically they should hit the proxy, which maintains only one connection to the camera itself. When there are no clients, the proxy does not connect to the camera. 

I don't think simple-rtsp-server is an issue, since the V3s work fine, and I tried only using the wyze app overnight (ie. not hitting the rtsp proxy) and it still crashed by morning.

The good news is in the latest build, it seems ethernet does work, but the boot up was unusually long (had the blinking blue light for a good few minutes), especially compared to the V3s which boot up relatively quickly. Not sure if its a related issue or not.

edit: it's unlikely its my network since SSH still works but in case...
My network is a gigabit connection to an ERX, which goes to a master switch. Wifi is driven by 2 unifi APs, one AC-LR and one AC-Pro. All the wifi MACs and ethernet adapters have static IPs assigned by the ERX

**Comment by clee604** [comment]
I rebooted the camera before i went to bed and didn't have any clients access the rtsp or wyze stream. When I woke up the video stream had crashed so I'm not sure if something is wrong with the rtsp server itself.

**Comment by clee604** [comment]
Update: I tried disabling all the options in the wyze app.

No SD recording, no timestamp, record sound off and so far the video feed has stayed up. I'm pretty sure it has crashed with the SD recording off before, so I wonder if the timestamp and/or record sound option is causing some issue.

If the stream stays up overnight, I'll try to isolate the option tomorrow by reenabling them one by one

**Comment by clee604** [comment]
Never mind, crashed this morning. Sigh, installed latest build to see if it makes any difference..

**Reply by gtxaspec** [reply]
does video in the app still work? or is it just rtsp

**Comment by clee604** [comment]
All video crashes including wyze app and rtsp. The newest build from this morning hasn't crashed.. Yet

**Comment by clee604** [comment]
Still hasn't crashed yet with newest build (July 9 2022), let's hope it has been fixed by whatever was released...

**Comment by gtxaspec** [comment]
keep us posted!

**Comment by clee604** [comment]
so, on this V2, I thought I was on ethernet with the bonding active but it seems it is falling back to wifi even though the ethernet is connected. Not sure if this has anything to do with the lack of crashes.... but I'm thinking the auto ethernet detection failed here so I'm going to manually set it and see if
a. ethernet works or not
b. does it crash with the 'fixed' settings

**Comment by clee604** [comment]
Still up, so looks like its resolved. The only thing is the DNS flood on the AWS servers which can be resolved using the command posted in a different thread, wish this could be automated on reboot though.
